reflection/reflect.go

```diff
@@ -2,40 +2,38 @@ package reflection
 
 import (
 	"errors"
+	stdlog "log"
 	"reflect"
 	"testing"
 	"unsafe"
 )
 
 // Gets a pointer of a private or public field in a testing.M struct
 func GetFieldPointerOfM(m *testing.M, fieldName string) (unsafe.Pointer, error) {
-	val := reflect.Indirect(reflect.ValueOf(m))
-	member := val.FieldByName(fieldName)
-	if member.IsValid() {
-		ptrToY := unsafe.Pointer(member.UnsafeAddr())
-		return ptrToY, nil
-	}
-	return nil, errors.New("field can't be retrieved")
+	return getFieldPointerOf(m, fieldName)
 }
 
 // Gets a pointer of a private or public field in a testing.T struct
 func GetFieldPointerOfT(t *testing.T, fieldName string) (unsafe.Pointer, error) {
-	val := reflect.Indirect(reflect.ValueOf(t))
-	member := val.FieldByName(fieldName)
-	if member.IsValid() {
-		ptrToY := unsafe.Pointer(member.UnsafeAddr())
-		return ptrToY, nil
-	}
-	return nil, errors.New("field can't be retrieved")
+	return getFieldPointerOf(t, fieldName)
 }
 
 // Gets a pointer of a private or public field in a testing.B struct
 func GetFieldPointerOfB(b *testing.B, fieldName string) (unsafe.Pointer, error) {
-	val := reflect.Indirect(reflect.ValueOf(b))
+	return getFieldPointerOf(b, fieldName)
+}
+
+// Gets a pointer of a private or public field in a testing.T struct
+func GetFieldPointerOfLogger(logger *stdlog.Logger, fieldName string) (unsafe.Pointer, error) {
+	return getFieldPointerOf(logger, fieldName)
+}
+
+func getFieldPointerOf(i interface{}, fieldName string) (unsafe.Pointer, error) {
```

Comment:
Can't we just make this public and use it and avoid the indirect `GetFieldPointerOfX`?

Reply:
I can do that if there is no problem to have that func public.

Reply:
I prefer to leave that func private and have control of exactly the structs we allow reflection.